### PR TITLE
drivers:adc added ADC_GAIN_3_4

### PR DESCRIPTION
**Pull request by erdemefe90**
ADC_GAIN_3_4 related things added adc-controller.yaml adc.h and adc_common.c

### CODE REVIEW

File: dts/bindings/adc/adc-controller.yaml [hunk 1]
```diff
@@ -50,6 +50,7 @@ child-binding:
           - ADC_GAIN_2_5: x 2/5
           - ADC_GAIN_1_2: x 1/2
           - ADC_GAIN_2_3: x 2/3
+          - ADC_GAIN_2_3: x 3/4
```
Comment (anangl) [inline dts/bindings/adc/adc-controller.yaml:53]:
```suggestion
          - ADC_GAIN_3_4: x 3/4
```

Comment (erdemefe90) [reply dts/bindings/adc/adc-controller.yaml]: fixed this line.

Comment (anangl) [reply dts/bindings/adc/adc-controller.yaml]: Please squash the two commits into one.

Comment (anangl) [reply dts/bindings/adc/adc-controller.yaml]: And correct the commit title so that it starts with "drivers: adc: ".

**Review by galak — DISMISSED** [review]

**Review by anangl — DISMISSED** [review]

**Review by henrikbrixandersen — CHANGES_REQUESTED** [review]
Please update `drivers/adc/adc_shell.c` as well.

**Pull request closed without merging**